spec/predicates/human-review/vcs.md

```diff
@@ -0,0 +1,103 @@
+# Predicate type: VCS Code Reviews
+
+Type URI: (tentative) https://in-toto.io/attestation/human-review/vcs/v0.1
+
+Version: 0.1.0
+
+## Purpose
+
+This attestation type is used to describe code review approvals of the simplest
+kind. The general expectation is that reviews are tied to specific changes in a
+version control system (VCS) such as Git.
+
+## Use Cases
+
+Changes to source code are reviewed by other members of the development team.
+The review may be performed using systems like Gerrit or via reviewing
+mechanisms built into version control forges such as GitHub and GitLab.
+
+## Prerequisite
+
+Understanding of the
+[in-toto attestation specification](https://github.com/in-toto/attestation).
+
+## Model
+
+Typically, software development processes incorporate some type of code review.
+Other members of the development team are expected to study the changes
+proposed and approve them, confirming they meet various expectations pertaining
+to code quality, functionality, and so on. Therefore, the general expectation
+is that this attestation is generated as a confirmation, an approval of some
+specific change, and is performed by developers in the supply chain who have
+code review responsibilities.
+
+## Schema
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [
+        {
+            "digest": { "gitCommit": "<HASH>" }
+        },
+    ],
+    "predicateType": "http://in-toto.io/attestation/human-review/vcs/v0.1",
+    "predicate": {
+        "result": "approved/changes-required/comment",
+        "reviewLink": "https://github.com/<>",
+        "timestamp": "<TIMESTAMP>"
+        "reviewer": "<STRING>"
+    }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the in-toto attestation parsing rules. Summary:
+
+-   Consumers MUST ignore unrecognized fields.
+-   The `predicateType` URI includes the major version number and will always
+    change whenever there is a backwards incompatible change.
+-   Minor version changes are always backwards compatible and “monotonic.” Such
+    changes do not update the `predicateType`.
+-   Producers MAY add extension fields using field names that are unlikely to
+    collide with names used by other producers. Field names SHOULD avoid using
+    characters like `.` and `$`.
+-   Fields marked _optional_ MAY be unset or null, and should be treated
+    equivalently. Both are equivalent to empty for object or array values.
+
+### Fields
+
+Extends: [human-review](./human-review.md#fields).
+
+`result` _enum_, _required_
+
+One of `approved`, `changes-requested`, or `comment`. TODO: This is inspired by
```

Comment:
What should folks assume based on these various fields?  What should a user do if they get one of these with the result of 'comment'?  If there are multiple attestations (because of multiple rounds of review) would the idea be "look for however many 'approved' statues from individual reviewers you require?"

Is it useful for this predicate type to cover reviews that are in-progress (vs those that are just complete and the code merged)?

Reply:
I thought about this for quite a while, my original draft had attestation generation only on approval (and monotonic). I was worried we'd lose some signal in the process though. Do you think it makes sense to go back to that model?

Also to answer the question more generally, I think assuming based on these fields should be in the realm of policies consuming the attestation. I'm fine with sketching some of that out as examples for the predicate spec once we nail it down some more.

Reply:
Would it make sense, given ITE-9, that this proposal should probably include some concrete use cases with example policies?

FWIW I'm not sure what I'd actually do with a 'changes-requested' review or a 'comment' review.